snuba/admin/static/tests/query_history.spec.tsx

```diff
@@ -0,0 +1,42 @@
+import { it, describe } from "@jest/globals";
+import { setRecentHistory, getRecentHistory } from "SnubaAdmin/query_history";
+
+describe("Query editor", () => {
+  global.ResizeObserver = require("resize-observer-polyfill");
+  describe("When setting history", () => {
+    it("should be retrievable from recentHistory", () => {
+      const TEST_KEY = "test"
+      setRecentHistory(TEST_KEY, {"something": "foo"});
+      expect(getRecentHistory(TEST_KEY)).toStrictEqual([{"something": "foo"}]);
+    });
+    it("inserts items at index 0", () => {
+      const TEST_KEY = "test1"
+      for (const index of [1,2]) {
+        setRecentHistory(TEST_KEY, {"something": index});
+      }
+      expect(getRecentHistory(TEST_KEY)).toStrictEqual([
+        {"something": 2},
+        {"something": 1},
+      ]);
+    });
+    it("drops the first item once a sixth item is pushed", () => {
+      const TEST_KEY = "test2"
+      for (const index of [1,2,3,4,5,6]) {
+        setRecentHistory(TEST_KEY, {"something": index});
+      }
+      expect(getRecentHistory(TEST_KEY)).toStrictEqual([
+        {"something": 6},
+        {"something": 5},
+        {"something": 4},
+        {"something": 3},
+        {"something": 2},
+      ]);
+    });
+    it("different keys don't conflict", () => {
+      const TEST_KEY1 = "test3"
+      const TEST_KEY2 = "test4"
+      setRecentHistory(TEST_KEY1, {"something": "foo"});
+      expect(getRecentHistory(TEST_KEY2)).toStrictEqual([]);
+    });
+  });
```

Comment:
Is it possible to add a test where `setRecentHistory()` raises exception and test confirms that `storageKey` is removed successfully from `localStorage`?

Reply:
Not easily, i think it'll add a lot of complexity to test that unfortunately. If that becomes a problem maybe its something we can revisit in the future